### DISCUSSION TITLE
Additional toggable POI layers (Booking.com and more)

### DISCUSSION THREAD

**Discussion by vbertola** [organicmaps/organicmaps#1391] [Ideas]
I know this is a big shot, but I'd like to drop for consideration something I've really missed from Maps.me in years of usage: a much bigger number of toggable POI layers.

For example, having all Booking.com structures on the map was useful when you had to find and pinpoint your newly booked B&B, but annoying when you were trying to navigate a city and all those structures would hide other types of POIs. So why not bring back Booking.com integration (legal issues apart - ping me separately if you want help on that) but only as an additional layer whose display you can turn on when you are looking for a structure, and turn off when you're done? (Any bookmarks associated to POIs on hidden layers should stay visible though.)

This would work also for many other types of POIs that are on OSM but are generally lost. For example, toilets: I'm looking for toilets but often I can't find them because they are hidden by other POIs nearby. Why don't we just put them to a separate layer that you can turn on (and in that case, it takes priority over the others) only when you need it? (by the way, the category search for toilets never really worked on Maps.me)

In the long term, this could even become an extension point for the app - third party developers could create a layer in a standard format and users could add layers from some kind of layer add-on menu.

**Comment by biodranik** [comment]
Do you have any ideas on how it can be implemented from a UX point of view?

Regarding Booking, a special deal should be made to save their data for offline usage. And there are many "apartments" that flood the map with garbage. So ideally everyone is interested in adding missing hotels into OSM directly instead of playing B2B games.

Btw, the best way to find a toilet is to use the search. The search even has a pre-filled category for it :)

**Comment by vbertola** [comment]
Users could just have more layers in the layer selection dialogue that appears when you click on the icon in the top left corner, which currently only offers terrain and metro lines. If you had too many possible layers, users could also get a menu page in the settings with a set of checkboxes, to decide which layers should appear in that menu.

Regarding the rest - yes, of course it would be better if hospitality structures were all added into OSM, but that's not going to happen, and also it's very hard to keep that information up to date (hotels change hands and names, many B&Bs open and close quickly etc). Booking platforms have a direct interest to manage that information and the money to do it, so they are going to be a much more reliable source than users editing OSM. (Ideally, one could also try to get Booking to adopt OSM and enter data directly there.)

And yes, I often tried the category search in Maps.me, it simply didn't work - somehow it offered me toilets 30 km away and missed the ones nearby.

**Reply by Misalf-git** [reply]
> And yes, I often tried the category search in Maps.me, it simply didn't work - somehow it offered me toilets 30 km away and missed the ones nearby.


Ordering of search results is weird. 

#1376
#1004

**Reply by biodranik** [reply]
We are thinking now about several "style" layers:
- City tourism
- Hiking
- Biking
- Driving

Making too many layers may complicate UX and development.

Making adding a missing hotel a fun experience could significantly improve OSM quality.

There are several bugs with the search, that hide nearby results from you. After fixing them, you don't need any layers for that task.

**Comment by pktpls** [comment]
I would be super interested in a map layer for our community mesh network, which would show local mesh sites as well as wifi links. Right now we have a web map based on Leaflet: https://hopglass.berlin.freifunk.net/

It could be quite similar to the public transport layer, except that there'd also be lots of points-of-interest having no direct links to others.
